### PR TITLE
Enable Mac Aarch64 reproducible comparing tests 

### PR DESCRIPTION
**Pull request by sophia-guo**

### CODE REVIEW

File: tooling/reproducible/repro_common.sh [hunk 1]
```diff
@@ -234,7 +237,7 @@ function removeMacOSNonComparableData() {
     else
       # Format uuid for BINREPL
       uuidhex="${uuid:0:2}:${uuid:2:2}:${uuid:4:2}:${uuid:6:2}:${uuid:8:2}:${uuid:10:2}:${uuid:12:2}:${uuid:14:2}:${uuid:16:2}:${uuid:18:2}:${uuid:20:2}:${uuid:22:2}:${uuid:24:2}:${uuid:26:2}:${uuid:28:2}:${uuid:30:2}"
-      if ! java "$TEMURIN_TOOLS_BINREPL" --inFile "$f" --outFile "$f" --hex "${uuidhex}-AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA" --firstOnly; then
+      if ! "${JDK_DIR}"/bin/java "$TEMURIN_TOOLS_BINREPL" --inFile "$f" --outFile "$f" --hex "${uuidhex}-AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA" --firstOnly; then
```
Comment (sophia-guo) [inline tooling/reproducible/repro_common.sh:240]: @andrew-m-leonard this line got the error 

```
00:18:45  Failed setting boot class path.
00:18:45    FAILED ==> java "temurin.tools.BinRepl" --inFile "src_jdk/Contents/Home/../../Contents/Home/bin/jwebserver" --outFile "src_jdk/Contents/Home/../../Contents/Home/bin/jwebserver" --hex "8C:04:09:F5:4F:35:3A:EB:B0:C1:42:BF:D8:09:D4:D5-AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA" --firstOnly
```

https://ci.adoptium.net/view/Test_grinder/job/Grinder/11059/console

Any suggestions?

Comment (andrew-m-leonard) [reply tooling/reproducible/repro_common.sh:240]: @sophia-guo it looks like the "java" being used is not working, is the PATH set correctly to find a useable JDK?
```
05:18:45  Error occurred during initialization of VM
05:18:45  Failed setting boot class path.
```

Comment (sophia-guo) [reply tooling/reproducible/repro_common.sh:242]: It was set to the tests jdk if ! "${JDK_DIR}"/bin/java" same as for the javap. Did you mean this is different from the javap, we can use whatever workable java? As I did notice that after we did the preprocess of the jdk ( remove signartures), java doesn't work any more.

**Review by karianna — APPROVED** [review]

**Review by karianna — CHANGES_REQUESTED** [review]
Will need a rebase

**Review by andrew-m-leonard — APPROVED** [review]
look ok I think

File: tooling/reproducible/repro_common.sh (hunk 1 above)
Comment (sophia-guo) [reply tooling/reproducible/repro_common.sh]: what's the required version of this useable JDK? Tried with tested jdk( jdk21)  got 
https://ci.adoptium.net/view/Test_grinder/job/Grinder/11124/console

```
20:52:57  Removing MacOS dylib non-comparable UUID from src_jdk/Contents/Home
20:53:17  Error: Could not find or load main class temurin.tools.BinRepl
20:53:17  Caused by: java.lang.ClassNotFoundException: temurin.tools.BinRepl
20:53:17    FAILED ==> /Users/admin/workspace/workspace/Grinder/aqa-tests/system/reproducibleCompare/comp-jdk-build/compare/src_jdk/Contents/Home_BK/bin/java "temurin.tools.BinRepl" --inFile "src_jdk/Contents/Home/../../Contents/Home/bin/jwebserver" --outFile "src_jdk/Contents/Home/../../Contents/Home/bin/jwebserver" --hex "8C:04:09:F5:4F:35:3A:EB:B0:C1:42:BF:D8:09:D4:D5-AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA" --firstOnly
```

**Review by karianna — APPROVED** [review]

**Review by karianna — CHANGES_REQUESTED** [review]
linter and license header updates needed

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/view/Test_grinder/job/Grinder/11167/console

@karianna updated.

**Review by karianna — APPROVED** [review]

File: tooling/reproducible/repro_common.sh (hunk 1 above)
Comment (sophia-guo) [reply tooling/reproducible/repro_common.sh]: Add a copy of compared jdk as the jdk to call javap. Everything looks good, except 150 libs files differs. FYI @andrew-m-leonard  https://ci.adoptium.net/view/Test_grinder/job/Grinder/11167/console

Comment (andrew-m-leonard) [reply tooling/reproducible/repro_common.sh]: @sophia-guo thank you, yes the console looks clean from a script perspective, which is great.
So the issue is now just making it 100%
I've raised the following issues on this and the other platforms:
- https://github.com/adoptium/temurin-build/issues/3988
- https://github.com/adoptium/temurin-build/issues/3989
- https://github.com/adoptium/temurin-build/issues/3990

**Pull request merged**